### PR TITLE
panda: refactor can_recv

### PR DESCRIPTION
**Pull request by deanlee**
unpack all data to the cache(`recv_buf)` at once

### CODE REVIEW

File: selfdrive/boardd/panda.cc [hunk 1]
```diff
@@ -425,64 +425,41 @@ void Panda::can_send(capnp::List<cereal::CanData>::Reader can_data_list) {
...
-
-  if (recv == RECV_SIZE) {
-    LOGW("Receive buffer full");
-  }
-
   if (!comms_healthy) {
     return false;
   }
-  return unpack_can_buffer(data, recv, out_vec);
+  if (recv == RECV_SIZE) { LOGW("Panda receive buffer full"); }
```
Comment (adeebshihadeh) [inline selfdrive/boardd/panda.cc:431]: use multiple lines here

File: selfdrive/boardd/panda.cc [hunk 2]
```diff
@@ -425,64 +425,41 @@ void Panda::can_send(capnp::List<cereal::CanData>::Reader can_data_list) {
...
+    canData.busTime = 0;
+    canData.address = header.addr;
+    canData.src = header.bus + bus_offset;
+    if (header.rejected) { canData.src += CANPACKET_REJECTED; }
+    if (header.returned) { canData.src += CANPACKET_RETURNED; }
+
+    const uint8_t data_len = dlc_to_len[header.data_len_code];
+    canData.dat.assign((char *)&recv_buf[pos + CANPACKET_HEAD_SIZE], data_len);
+
+    pos += CANPACKET_HEAD_SIZE + data_len;
```
Comment (adeebshihadeh) [inline selfdrive/boardd/panda.cc:462]: move this into the for?

Comment (briskspirit) [reply selfdrive/boardd/panda.cc:464]: ?

Comment (briskspirit) [reply selfdrive/boardd/panda.cc:464]: No difference but don't need to declare pos on extra line. Think that's why he chose it?

**Pull request merged**